File: frontend/src/api/ai.ts
import { apiClient, handleApiError } from './client';

export interface ChatMessage {
  id: string;
  message_role: 'user' | 'assistant';
  content: string;
  created_at: string;
}

export const getChatHistory = async (userId: string): Promise<ChatMessage[]> => {
  try {
    const response = await apiClient.get(`/ai/history/${userId}`);
    return response.data;
  } catch (error) {
    throw new Error(handleApiError(error));
  }
};

export const sendChatMessage = async (userId: string, portfolioId: string, message: string): Promise<{response: string, tool_calls_made: string[]}> => {
  try {
    const response = await apiClient.post('/ai/chat', {
      user_id: userId,
      portfolio_id: portfolioId,
      message: message
    });
    return response.data;
  } catch (error) {
    throw new Error(handleApiError(error));
  }
};

File: backend/app/schemas/ai.py
from pydantic import BaseModel
from typing import Optional, List
import uuid
from datetime import datetime

class ChatMessage(BaseModel):
    role: str # "user" or "assistant"
    content: str

class ChatRequest(BaseModel):
    user_id: uuid.UUID
    portfolio_id: uuid.UUID
    message: str

class ChatResponse(BaseModel):
    response: str
    tool_calls_made: List[str] = []

class ChatHistorySchema(BaseModel):
    id: uuid.UUID
    message_role: str
    content: str
    created_at: datetime

    class Config:
        from_attributes = True
